components/core/src/os/users/linux.rs

```diff
@@ -26,46 +26,48 @@ pub fn can_run_services_as_svc_user() -> bool {
 pub fn can_run_services_as_svc_user() -> bool { true }
 
 pub fn get_uid_by_name(owner: &str) -> Option<u32> {
-    users::get_user_by_name(owner).map(|u| u.uid())
+    User::from_name(owner).ok()
+                          .flatten()
```

Comment:
The use of `flatten` here (and elsewhere in this PR) surprised me a bit, but it's pretty clever.

However, I'm not sure if it's a great idea to swallow this error like this. This might be a good place to use the ["lognore"](https://github.com/habitat-sh/habitat/pull/6794#discussion_r311768892) pattern that has been discussed a few times, but not yet implemented.

Reply:
@christophermaier I have a follow up change that changes these functions to all return Result<Option<T>> so that we don't have to ignore the change.

But, now that you mention it, I think I like logging on error here better because the story around errors out of the underlying functions is a bit more complicated than I would like.

Reply:
Addressed in https://github.com/habitat-sh/habitat/pull/7620/commits/4b068a3f34e9ba5fcecacbefdf0d84dfa11e9f36